frontend/src/components/Obligatory.jsx

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable no-alert */
+import React, { useState } from "react";
+import Essential from "./Essential";
+
+function Obligatory() {
+  const [question1, setQuestion1] = useState("");
+  const [question2, setQuestion2] = useState("");
+  const [question3, setQuestion3] = useState("");
+  const [question4, setQuestion4] = useState("");
+  const [obligatoryCompleted, setObligatoryCompleted] = useState(false);
+
+  const handleQuestion1Change = (event) => {
+    setQuestion1(event.target.value);
+  };
+
+  const handleQuestion2Change = (event) => {
+    setQuestion2(event.target.value);
+  };
+
+  const handleQuestion3Change = (event) => {
+    setQuestion3(event.target.value);
+  };
+
+  const handleQuestion4Change = (event) => {
+    setQuestion4(event.target.value);
+  };
+
+  const handleNextSection = () => {
+    if (question1 && question2 && question3 && question4) {
+      setObligatoryCompleted(true);
+    } else {
+      alert("Veuillez répondre à toutes les questions obligatoires");
+    }
+  };
+
+  return (
```

Comment:
you should not have an Obligatory, Essential and an Optional component, but a Question component with a `type` prop